File: OOP_final_project_JAVA_version/src/exercicio_incremental_parte2/Contrato.java
package exercicio_incremental_parte2;
import java.util.ArrayList;
import java.util.Date;
import java.util.Iterator;

public class Contrato {
    private int num_contrato;
    private Date data_inicio;
    private Date data_fim;
    private double valor;
    //private Motorista motorista;
    private ArrayList<Motorista> motorista;
    //private Veiculo veiculo;
    private ArrayList<Veiculo> veiculo;
    private ArrayList<Fonecedor>fornecedor;
    

    public Contrato(int num_contrato, Date data_inicio, Date data_fim, double valor) {
        this.num_contrato = num_contrato;
        this.data_inicio = data_inicio;
        this.data_fim = data_fim;
        this.valor = valor;
        this.veiculo = new ArrayList<>();
        this.motorista = new ArrayList<>();
    }
    
    //metodo para adicionar veiculo
    public String alteraVeiculo(Veiculo veiculo) {
        if(veiculo.getTipo()==1){
            this.veiculo.add(veiculo);
            return "veiculo adicionado com succeso";
        }else{
            return "veiculo proprio nao pode ser adicionado";
        }
        
    }
    //metodo para remover veiculo
    public String alteraVeiculo(String placa){
        for(Iterator<Veiculo>it = veiculo.iterator(); it.hasNext();){
            Veiculo x = it.next();
            if(x.getPlaca()==placa){
                this.veiculo.remove(x);
                return "Veiculo removido com sucesso";                
            }else{
                return "Veiculo nao consta no contrato para ser removido";
            }
        }
        return "";
    }
    
    public void adicionaFornecedor(Fonecedor fornecedor){
        this.fornecedor.add(fornecedor);
    }
    
    public String alteraMotorista(Motorista motorista) {
        if(motorista.getTipo()==1){
            this.motorista.add(motorista);
            return "motorista adicionado com succeso";
        }else{
            return "motorista servidor nao pode ser adicionado";
        }
        
    }
    
    public String alteraMotorista(String cpf_cnpj){
        for(Iterator<Motorista>it = motorista.iterator(); it.hasNext();){
            Motorista x = it.next();
            if(x.getCpf_cnpj()==cpf_cnpj){
                this.motorista.remove(x);
                return "Motorista removido com sucesso";                
            }else{
                return "Motorista nao consta no contrato para ser removido";
            }
        }
        return "";
    }

    public int getNum_contrato() {
        return num_contrato;
    }

    public void setNum_contrato(int num_contrato) {
        this.num_contrato = num_contrato;
    }

    public Date getData_inicio() {
        return data_inicio;
    }

    public void setData_inicio(Date data_inicio) {
        this.data_inicio = data_inicio;
    }

    public Date getData_fim() {
        return data_fim;
    }

    public void setData_fim(Date data_fim) {
        this.data_fim = data_fim;
    }

    public double getValor() {
        return valor;
    }

    public void setValor(double valor) {
        this.valor = valor;
    }
    
    
}

File: OOP_final_project_python_version/contrato.py
from datetime import date
from motorista import Motorista
#from veiculo import Veiculo
from fornecedor import Fornecedor
from pessoa_juridica import PessoaJuridica

class Contrato:
    def __init__(self, num_contrato, data_inicio, data_fim, valor):
        self.__num_contrato = num_contrato
        self.__data_inicio = data_inicio
        self.__data_fim = data_fim
        self.__valor = valor
        self.__veiculo = []
        self.__motorista = []
        self.__fornecedor = []

    @property
    def num_contrato(self):
        return self.__num_contrato

    @num_contrato.setter
    def num_contrato(self, num_contrato):
        self.__num_contrato = num_contrato

    @property
    def data_inicio(self):
        return self.__data_inicio

    @data_inicio.setter
    def data_inicio(self, data_inicio):
        self.__data_inicio = data_inicio

    @property
    def data_fim(self):
        return self.__data_fim

    @data_fim.setter
    def data_fim(self, data_fim):
        self.__data_fim = data_fim

    @property
    def valor(self):
        return self.__valor

    @valor.setter
    def valor(self, valor):
        self.__valor = valor

    def altera_veiculo(self, veiculo):
        if veiculo.tipo == 1:
            self.__veiculo.append(veiculo)
            return "Veículo adicionado com sucesso"
        else:
            return "Veículo próprio não pode ser adicionado"
    '''
    def remove_veiculo(self, placa):
        for v in self.__veiculo:
            if v.placa == placa:
                self.__veiculo.remove(v)
                return "Veículo removido com sucesso"
        return "Veículo não consta no contrato para ser removido"
    '''

    #so adciona se for forncedor
    
    def adiciona_pessoa_juridica(self, pessoa_juridica):
            if isinstance(pessoa_juridica, PessoaJuridica) and isinstance(pessoa_juridica, Fornecedor):
                self.__fornecedor.append(pessoa_juridica)
                return "Pessoa Jurídica (Fornecedor) adicionada com sucesso ao contrato"
            else:
                return "Erro: O objeto fornecido não é uma Pessoa Jurídica (Fornecedor) válida."


    def adiciona_fornecedor(self, fornecedor):
        self.__fornecedor.append(fornecedor)

    def altera_motorista(self, motorista):
        if motorista.tipo == 1:
            self.__motorista.append(motorista)
            return "Motorista adicionado com sucesso"
        else:
            return "Motorista servidor não pode ser adicionado"

    def remove_motorista(self, cpf_cnpj):
        for m in self.__motorista:
            if m.cpf_cnpj == cpf_cnpj:
                self.__motorista.remove(m)
                return "Motorista removido com sucesso"
        return "Motorista não consta no contrato para ser removido"
    
    